provider/kubeletconfig/kubeletconfig_resource.go

```diff
@@ -163,10 +199,25 @@ func (k *KubeletConfigResource) Read(ctx context.Context, req resource.ReadReque
 	}
 
 	clusterId := state.Cluster.ValueString()
-	exists, kubeletConfig, err := k.configClient.Exists(ctx, clusterId)
+
+	var kubeletConfigId string
+	var err error
+	if common.IsStringAttributeUnknownOrEmpty(state.ID) {
```

Comment:
Could we move this check into a common function since its being used in 3 different instances?

Reply:
made the refactoring, please re-review again, thanks